src/utils/decorators/tooltip-decorator/tooltip-decorator.js

```diff
@@ -83,52 +85,17 @@ import { styleElement, append } from '../../ether';
 const TooltipDecorator = (ComposedComponent) => {
   class Component extends ComposedComponent {
     static propTypes = assign({}, ComposedComponent.propTypes, {
-
-      /**
-       * The message for this tooltip
-       *
-       * @property
-       * @type {Node}
-       */
       tooltipMessage: PropTypes.node,
-
-      /**
-       * The position of this tooltip: top, bottom, left or right
-       *
-       * @property
-       * @default top
-       * @type {String}
-       */
-      tooltipPosition: PropTypes.string,
-
-      /**
-       * The alignment of this tooltip: left, right or center
-       *
-       * @property
-       * @default center
-       * @type {String}
-       */
-      tooltipAlign: PropTypes.string
+      tooltipPosition: PropTypes.oneOf(OptionsHelper.positions),
+      tooltipAlign: PropTypes.oneOf(OptionsHelper.alignAroundEdges)
     });
 
-    /**
-     * Timeout for firing ajax request for showing the tooltip
-     *
-     * @property _showTooltipTimeout
-     */
     _showTooltipTimeout = null;
 
-    /**
-     * Timeout for firing ajax request for hiding the tooltip
-     *
-     * @property _hideTooltipTimeout
-     */
     _hideTooltipTimeout = null;
 
     /**
      * Cache the shifts calculations (used for positioning)
-     *
-     * @property _memoizedShifts
      */
     _memoizedShifts = null;
```

Comment:
Currently, this `_memoizedShifts` property doesn't seem to do anything useful (there's nowhere in the code where `_memoizedShifts` is set to a value other than `null`).

In my [`FE-1847-tooltip-positioning`](https://github.com/Sage/carbon/commits/FE-1847-tooltip-positioning) branch, I deleted `_memoizedShifts`, and instead I implemented an optimisation where `styleElement()` is only called if the CSS attribute's value has changed – see [`setPositionAttributeIfChanged()` in `tooltip-position.config.js`](https://github.com/Sage/carbon/blob/1a0afac/src/components/tooltip/tooltip-position.config.js#L11-L16).

Reply:
Good poiunt, I have implemented a solution directly in the `styleElement` method

Reply:
Thanks for removing `_memoizedShifts`, and adding the optimisation in `styleElement()` in `src/utils/ether/ether.js` :white_check_mark: